### ISSUE TITLE
Make traits like `StringOrFunction` public

### ISSUE THREAD

**Issue by ModProg** [noib3/nvim-oxi#50]
It would be good to either make them public or improve the documentation everywhere they are used, so that you can have a quick look on what actually are the restrictions on the function e.g. here: https://docs.rs/nvim-oxi/latest/nvim_oxi/api/fn.create_user_command.html where you cannot click on `StringOrFunction`.

**Comment by noib3** [comment]
I agree we should expand the documentation but I'd leave them private for now.
